src/libraries/Descriptor.sol

```diff
@@ -0,0 +1,520 @@
+    }
+
+    function addressToString(address addr) internal pure returns (string memory) {
+        return (uint256(uint160(addr))).toHexString(20);
+    }
+
+    /// @notice Generates the SVG image for a Uniswap V4 NFT
+    /// @param params Parameters needed to generate the SVG image
+    /// @return svg The SVG image as a string
+    function generateSVGImage(ConstructTokenURIParams memory params) internal pure returns (string memory svg) {
+        SVG.SVGParams memory svgParams = SVG.SVGParams({
+            quoteCurrency: addressToString(Currency.unwrap(params.quoteCurrency)),
+            baseCurrency: addressToString(Currency.unwrap(params.baseCurrency)),
+            hooks: params.hooks,
+            quoteCurrencySymbol: params.quoteCurrencySymbol,
+            baseCurrencySymbol: params.baseCurrencySymbol,
+            feeTier: feeToPercentString(params.fee),
+            tickLower: params.tickLower,
+            tickUpper: params.tickUpper,
+            tickSpacing: params.tickSpacing,
+            overRange: overRange(params.tickLower, params.tickUpper, params.tickCurrent),
+            tokenId: params.tokenId,
+            color0: currencyToColorHex(params.quoteCurrency.toId(), 136),
+            color1: currencyToColorHex(params.baseCurrency.toId(), 136),
+            color2: currencyToColorHex(params.quoteCurrency.toId(), 0),
+            color3: currencyToColorHex(params.baseCurrency.toId(), 0),
+            x1: scale(getCircleCoord(params.quoteCurrency.toId(), 16, params.tokenId), 0, 255, 16, 274),
+            y1: scale(getCircleCoord(params.baseCurrency.toId(), 16, params.tokenId), 0, 255, 100, 484),
+            x2: scale(getCircleCoord(params.quoteCurrency.toId(), 32, params.tokenId), 0, 255, 16, 274),
+            y2: scale(getCircleCoord(params.baseCurrency.toId(), 32, params.tokenId), 0, 255, 100, 484),
+            x3: scale(getCircleCoord(params.quoteCurrency.toId(), 48, params.tokenId), 0, 255, 16, 274),
+            y3: scale(getCircleCoord(params.baseCurrency.toId(), 48, params.tokenId), 0, 255, 100, 484)
+        });
+
+        return SVG.generateSVG(svgParams);
+    }
+
+    /// @notice Checks if the current price is within your position range, above, or below
+    /// @param tickLower The lower tick
+    /// @param tickUpper The upper tick
+    /// @param tickCurrent The current tick
+    /// @return 0 if the current price is within the position range, -1 if below, 1 if above
+    function overRange(int24 tickLower, int24 tickUpper, int24 tickCurrent) private pure returns (int8) {
+        if (tickCurrent < tickLower) {
+            return -1;
+        } else if (tickCurrent > tickUpper) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    function scale(uint256 n, uint256 inMn, uint256 inMx, uint256 outMn, uint256 outMx)
+        private
+        pure
+        returns (string memory)
+    {
+        return (n - inMn * (outMx - outMn) / (inMx - inMn) + outMn).toString();
```

Comment:
Is this still equivalent to before?

```
n.sub(inMn).mul(outMx.sub(outMn)).div(inMx.sub(inMn)).add(outMn)).toString();
```

I feel like before the beginning of it was
`(n - inMn) * x`
and now because of lack of bracketing wont it be executed as
`(n - (inMin * x)`

Reply:
good catch